src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/_params.py

```diff
@@ -0,0 +1,139 @@
+                      help='The subject name of the certificate to be created.')
+register_cli_argument('sf', 'vault_resource_group_name',options_list=('--vault-resource-group', '-vg','-vrg'),
+                      help='Azure key vault resource group name, it not given it will be defaulted to cluster resource group name')
+register_cli_argument('sf', 'vault_name',options_list=('--vault-name', '-vn'),
+                      help='Azure key vault name, it not given it will be defaulted to the cluster resource group name')
+register_cli_argument('sf', 'cluster_size',options_list=('--cluster-size', '-size','-s'),
+                      help='The number of nodes in the cluster. Default are 5 nodes')
+register_cli_argument('sf', 'vm_sku', options_list=('--vm-sku', '-sku'), help='The Vm Sku')
+register_cli_argument('sf', 'vm_user_name', options_list=('--vm-user-name', '-un'), help='The user name for logging to Vm. Default will be adminuser')
+register_cli_argument('sf', 'vm_os', default='WindowsServer2016Datacenter', options_list=('--vm-os', '-os'),
+                      help='The Operating System of the VMs that make up the cluster.',
+                     **enum_choice_list(['WindowsServer2012R2Datacenter', 'WindowsServer2016Datacenter', 'WindowsServer2016DatacenterwithContainers', 'UbuntuServer1604']))
+
+register_cli_argument('sf client certificate', 'certificate_common_name',
+                      help='Specify client certificate common name.')
+register_cli_argument('sf client certificate', 'admin_client_thumbprints',
+                      help='Specify list of client certificate thumbprint that only has admin permission.')
+register_cli_argument('sf client certificate', 'certificate_issuer_thumbprint',
+                      help='Specify client certificate issuer thumbprint.')
+
+register_cli_argument('sf cluster client-certificate', 'is_admin',
+                      help='Client authentication type.')
+register_cli_argument('sf cluster client-certificate', 'certificate_issuer_thumbprint',
+                      help='Specify client certificate issuer thumbprint.')
+register_cli_argument('sf cluster client-certificate', 'certificate_common_name',
+                      help='Specify client certificate common name.')
+register_cli_argument('sf cluster client-certificate', 'admin_client_thumbprints',
+                      help='Specify client certificate thumbprint that only has admin permission.')
+register_cli_argument('sf cluster client-certificate', 'readonly_client_thumbprints',
+                      help='Specify list of client certificate thumbprint that has read only permission.') 
+register_cli_argument('sf cluster client-certificate add', 'thumbprint',
+                      help='Specify client certificate thumbprint.')
+register_cli_argument('sf cluster client-certificate remove', 'thumbprint',
+                      help='Specify a single or list of client certificate thumbprint(s) to be remove.')
```

Comment:
What is the actual type the SDK accepts?  Unless it is a string, this will fail. 
`--thumbprint ['abc', 'def']` will definitely not work on our supported shells.